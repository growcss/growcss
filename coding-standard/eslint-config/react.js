module.exports ={
  extends: [
    'prettier/react',
    'plugin:jest/recommended'
  ],
  plugins: [
    'react',
    'jsx-a11y',
    // 'security', @todo check it on the next version
  ],
  rules: {
    ...{
      'react/sort-comp': 'off',
      'react/jsx-filename-extension': 'off',
      'react/require-default-props': 'off',
    },
    ...require('./rules/prettier-conflicts-react'),
    ...require('./rules/jest'),
  },
  env: {
    browser: true,
    commonjs: true,
    es6: true,
    jest: true,
    node: true,
  },
};
