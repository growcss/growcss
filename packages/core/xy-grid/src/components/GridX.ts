import AbstractGrid from './AbstractGrid';

export default class GridX extends AbstractGrid {
  /**
   * @inheritdoc
   */
  vertical: boolean = false;
}
