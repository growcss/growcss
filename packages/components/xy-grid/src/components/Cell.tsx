import * as React from 'react';
import classNames from 'classnames';
import { Gutters as DefaultGutters } from './Gutters';
import { CellElement } from '../styled/CellElement';

export interface CellProps {
    children: React.Node,
    gridColumns?: number,
    vertical?: boolean,
    gutterSizes?: string | number | GuttersType,
    cellType?: string,
    gutterType?: string,
    small?: number | string,
    smallOffset?: number,
    medium?: number | string,
    mediumOffset?: number,
    large?: number | string,
    largeOffset?: number,
    xlarge?: number | string,
    xlargeOffset?: number,
    xxlarge?: number | string,
    xxlargeOffset?: number,
    align?: string,
}

export default class Cell extends React.Component<CellProps> {
  static defaultProps = {
    gridColumns: 12,
    gutterSizes: DefaultGutters,
    vertical: false,
  };

  render() {
    const {
      children,
      gridColumns,
      cellType,
      gutterType,
      gutterSizes,
      vertical,
      small,
      smallOffset,
      medium,
      mediumOffset,
      large,
      largeOffset,
      xlarge,
      xlargeOffset,
      xxlarge,
      xxlargeOffset,
      align,
      ...other
    } = this.props;
    const className = classNames('gc-cell');

    return (
      <CellElement
        className={className}
        gridColumns={gridColumns}
        cellType={cellType}
        gutterType={gutterType}
        gutterSizes={gutterSizes}
        vertical={vertical}
        small={small}
        smallOffset={smallOffset}
        medium={medium}
        mediumOffset={mediumOffset}
        large={large}
        largeOffset={largeOffset}
        xlarge={xlarge}
        xlargeOffset={xlargeOffset}
        xxlarge={xxlarge}
        xxlargeOffset={xxlargeOffset}
        align={align}
        {...other}
      >
        {children}
      </CellElement>
    );
  }
}
