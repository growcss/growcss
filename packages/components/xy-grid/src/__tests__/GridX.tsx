import * as React from 'react';
import GridX from '../components/GridX';
import Cell from '../components/Cell';
import 'jest-styled-components';
import { shallow } from 'enzyme';

test('the style output of GridX', () => {
  const wrapper = shallow(
    <GridX>
      <Cell>test</Cell>
    </GridX>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('display', 'flex');
  expect(wrapper).toHaveStyleRule('flex-flow', 'row wrap');
  expect(wrapper).toHaveStyleRule('-webkit-box-orient', 'horizontal');
  expect(wrapper).toHaveStyleRule('-webkit-box-direction', 'normal');
  expect(wrapper).toHaveStyleRule('justify-content', 'flex-start');
});

test('the style output of GridX with alignment to right', () => {
  const wrapper = shallow(
    <GridX gutterType="padding" alignX="right">
      <Cell>test</Cell>
    </GridX>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('display', 'flex');
  expect(wrapper).toHaveStyleRule('flex-flow', 'row wrap');
  expect(wrapper).toHaveStyleRule('-webkit-box-orient', 'horizontal');
  expect(wrapper).toHaveStyleRule('-webkit-box-direction', 'normal');
  expect(wrapper).toHaveStyleRule('justify-content', 'flex-end');
});

test('the style output of GridX with alignment to center', () => {
  const wrapper = shallow(
    <GridX gutterType="padding" alignX="center">
      <Cell>test</Cell>
    </GridX>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('display', 'flex');
  expect(wrapper).toHaveStyleRule('flex-flow', 'row wrap');
  expect(wrapper).toHaveStyleRule('-webkit-box-orient', 'horizontal');
  expect(wrapper).toHaveStyleRule('-webkit-box-direction', 'normal');
  expect(wrapper).toHaveStyleRule('justify-content', 'center');
});

test('the style output of GridX with alignment to justify', () => {
  const wrapper = shallow(
    <GridX gutterType="padding" alignX="justify">
      <Cell>test</Cell>
    </GridX>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('display', 'flex');
  expect(wrapper).toHaveStyleRule('flex-flow', 'row wrap');
  expect(wrapper).toHaveStyleRule('-webkit-box-orient', 'horizontal');
  expect(wrapper).toHaveStyleRule('-webkit-box-direction', 'normal');
  expect(wrapper).toHaveStyleRule('justify-content', 'space-between');
});

test('the style output of GridX with alignment to spaced', () => {
  const wrapper = shallow(
    <GridX gutterType="padding" alignX="spaced">
      <Cell>test</Cell>
    </GridX>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('display', 'flex');
  expect(wrapper).toHaveStyleRule('flex-flow', 'row wrap');
  expect(wrapper).toHaveStyleRule('-webkit-box-orient', 'horizontal');
  expect(wrapper).toHaveStyleRule('-webkit-box-direction', 'normal');
  expect(wrapper).toHaveStyleRule('justify-content', 'space-around');
});

test('the style output of GridX with alignment to top', () => {
  const wrapper = shallow(
    <GridX gutterType="padding" alignY="top">
      <Cell>test</Cell>
    </GridX>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('display', 'flex');
  expect(wrapper).toHaveStyleRule('flex-flow', 'row wrap');
  expect(wrapper).toHaveStyleRule('-webkit-box-orient', 'horizontal');
  expect(wrapper).toHaveStyleRule('-webkit-box-direction', 'normal');
  expect(wrapper).toHaveStyleRule('align-items', 'flex-start');
});

test('the style output of GridX with alignment to middle', () => {
  const wrapper = shallow(
    <GridX gutterType="padding" alignY="middle">
      <Cell>test</Cell>
    </GridX>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('display', 'flex');
  expect(wrapper).toHaveStyleRule('flex-flow', 'row wrap');
  expect(wrapper).toHaveStyleRule('-webkit-box-orient', 'horizontal');
  expect(wrapper).toHaveStyleRule('-webkit-box-direction', 'normal');
  expect(wrapper).toHaveStyleRule('align-items', 'center');
});

test('the style output of GridX with alignment to bottom', () => {
  const wrapper = shallow(
    <GridX gutterType="padding" alignY="bottom">
      <Cell>test</Cell>
    </GridX>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('display', 'flex');
  expect(wrapper).toHaveStyleRule('flex-flow', 'row wrap');
  expect(wrapper).toHaveStyleRule('-webkit-box-orient', 'horizontal');
  expect(wrapper).toHaveStyleRule('-webkit-box-direction', 'normal');
  expect(wrapper).toHaveStyleRule('align-items', 'flex-end');
});

test('the style output of GridX with alignment to stretch', () => {
  const wrapper = shallow(
    <GridX gutterType="padding" alignY="stretch">
      <Cell>test</Cell>
    </GridX>,
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('display', 'flex');
  expect(wrapper).toHaveStyleRule('flex-flow', 'row wrap');
  expect(wrapper).toHaveStyleRule('-webkit-box-orient', 'horizontal');
  expect(wrapper).toHaveStyleRule('-webkit-box-direction', 'normal');
  expect(wrapper).toHaveStyleRule('align-items', 'stretch');
});
