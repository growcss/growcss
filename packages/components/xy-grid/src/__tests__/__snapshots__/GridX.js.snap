// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the style output of GridX 1`] = `
<XYGridElement
  alignX="left"
  className="gc-grid-x"
  gridDirection="horizontal"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  wrap={true}
>
  <Cell
    gridColumns={12}
    gutterSizes={
      Object {
        "medium": "30px",
        "small": "20px",
      }
    }
    key=".0"
    vertical={false}
  >
    test
  </Cell>
</XYGridElement>
`;

exports[`the style output of GridX with alignment to bottom 1`] = `
<XYGridElement
  alignX="left"
  alignY="bottom"
  className="gc-grid-x"
  gridDirection="horizontal"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  wrap={true}
>
  <Cell
    gridColumns={12}
    gutterSizes={
      Object {
        "medium": "30px",
        "small": "20px",
      }
    }
    gutterType="padding"
    key=".0"
    vertical={false}
  >
    test
  </Cell>
</XYGridElement>
`;

exports[`the style output of GridX with alignment to center 1`] = `
<XYGridElement
  alignX="center"
  className="gc-grid-x"
  gridDirection="horizontal"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  wrap={true}
>
  <Cell
    gridColumns={12}
    gutterSizes={
      Object {
        "medium": "30px",
        "small": "20px",
      }
    }
    gutterType="padding"
    key=".0"
    vertical={false}
  >
    test
  </Cell>
</XYGridElement>
`;

exports[`the style output of GridX with alignment to justify 1`] = `
<XYGridElement
  alignX="justify"
  className="gc-grid-x"
  gridDirection="horizontal"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  wrap={true}
>
  <Cell
    gridColumns={12}
    gutterSizes={
      Object {
        "medium": "30px",
        "small": "20px",
      }
    }
    gutterType="padding"
    key=".0"
    vertical={false}
  >
    test
  </Cell>
</XYGridElement>
`;

exports[`the style output of GridX with alignment to middle 1`] = `
<XYGridElement
  alignX="left"
  alignY="middle"
  className="gc-grid-x"
  gridDirection="horizontal"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  wrap={true}
>
  <Cell
    gridColumns={12}
    gutterSizes={
      Object {
        "medium": "30px",
        "small": "20px",
      }
    }
    gutterType="padding"
    key=".0"
    vertical={false}
  >
    test
  </Cell>
</XYGridElement>
`;

exports[`the style output of GridX with alignment to right 1`] = `
<XYGridElement
  alignX="right"
  className="gc-grid-x"
  gridDirection="horizontal"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  wrap={true}
>
  <Cell
    gridColumns={12}
    gutterSizes={
      Object {
        "medium": "30px",
        "small": "20px",
      }
    }
    gutterType="padding"
    key=".0"
    vertical={false}
  >
    test
  </Cell>
</XYGridElement>
`;

exports[`the style output of GridX with alignment to spaced 1`] = `
<XYGridElement
  alignX="spaced"
  className="gc-grid-x"
  gridDirection="horizontal"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  wrap={true}
>
  <Cell
    gridColumns={12}
    gutterSizes={
      Object {
        "medium": "30px",
        "small": "20px",
      }
    }
    gutterType="padding"
    key=".0"
    vertical={false}
  >
    test
  </Cell>
</XYGridElement>
`;

exports[`the style output of GridX with alignment to stretch 1`] = `
<XYGridElement
  alignX="left"
  alignY="stretch"
  className="gc-grid-x"
  gridDirection="horizontal"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  wrap={true}
>
  <Cell
    gridColumns={12}
    gutterSizes={
      Object {
        "medium": "30px",
        "small": "20px",
      }
    }
    gutterType="padding"
    key=".0"
    vertical={false}
  >
    test
  </Cell>
</XYGridElement>
`;

exports[`the style output of GridX with alignment to top 1`] = `
<XYGridElement
  alignX="left"
  alignY="top"
  className="gc-grid-x"
  gridDirection="horizontal"
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  wrap={true}
>
  <Cell
    gridColumns={12}
    gutterSizes={
      Object {
        "medium": "30px",
        "small": "20px",
      }
    }
    gutterType="padding"
    key=".0"
    vertical={false}
  >
    test
  </Cell>
</XYGridElement>
`;
